src/TelemetryChannels/ServerTelemetryChannel/Shared/SamplingTelemetryProcessor.cs

```diff
@@ -26,8 +41,71 @@ public SamplingTelemetryProcessor(ITelemetryProcessor next)
 
             this.SamplingPercentage = 100.0;
             this.Next = next;
+            this.excludedTypesHashSet = new HashSet<Type>();
         }
-        
+
+        /// <summary>
+        /// Gets or sets a semicolon separated list of telemetry types that should not be sampled.
+        /// </summary>
+        public string ExcludedTypes
+        {
+            get
+            {
+                return this.excludedTypesString;
+            }
+
+            set
+            {
+                this.excludedTypesString = value;
+
+                HashSet<Type> newExcludedTypesHashSet = new HashSet<Type>();
+                if (!string.IsNullOrEmpty(value))
+                {
+                    string[] splitList = value.Split(this.listSeparators, StringSplitOptions.RemoveEmptyEntries);
+                    foreach (string item in splitList)
+                    {
+                        if (string.Compare(item, DependencyTelemetryName, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            newExcludedTypesHashSet.Add(typeof(DependencyTelemetry));
+                            continue;
+                        }
+
+                        if (string.Compare(item, EventTelemetryName, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            newExcludedTypesHashSet.Add(typeof(EventTelemetry));
+                            continue;
+                        }
+
+                        if (string.Compare(item, ExceptionTelemetryName, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            newExcludedTypesHashSet.Add(typeof(ExceptionTelemetry));
+                            continue;
+                        }
+
+                        if (string.Compare(item, PageViewTelemetryName, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            newExcludedTypesHashSet.Add(typeof(PageViewTelemetry));
+                            continue;
+                        }
+
+                        if (string.Compare(item, RequestTelemetryName, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            newExcludedTypesHashSet.Add(typeof(RequestTelemetry));
+                            continue;
+                        }
+
+                        if (string.Compare(item, TraceTelemetryName, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            newExcludedTypesHashSet.Add(typeof(TraceTelemetry));
+                            continue;
+                        }
```

Comment:
I do not need to deal with duplicates and it would work faster on lookup since I do not need to do ToString

---
In reply to: [67455785](https://github.com/Microsoft/ApplicationInsights-dotnet/pull/243#discussion_r67455785) [](ancestors = 67455785)